### ISSUE TITLE
A better error for writing functions with upper-case first letter

### ISSUE THREAD

**Issue by martinmodrak** [elm-lang/error-message-catalog#163]
This simple program

```
A : Float
A = 10
```

gives this error:

```
I ran into something unexpected when parsing your code!

1| A : Float
     ^
I am looking for one of the following things:

    an equals sign '='
    whitespace
```

I guess it should say something as:

```
1| A : Float
     ^
Names of functions/values cannot start with an upper-case letter.
```

**Comment by folkertdev** [comment]
there's more to this. The error @martincerny  gets only happens when there  is only one type constructor and the type signature is given. 

[Here are some more weird cases](https://github.com/elm-lang/elm-compiler/issues/1486)

**Comment by evancz** [comment]
Thank you for the report! I am finishing up an overhaul of the parser that will produce nicer error messages. I made a tweak based on your example, so my development build is now producing the following message:

```
-- SYNTAX PROBLEM ----------------------------------------------------- temp.elm

I was expecting to see a declaration, but I am getting stuck here:

1| A : Float
   ^
All declarations start with lower-case letters in Elm, so maybe this letter can
be changed to lower-case?

Hint: Here are a couple valid declarations for reference:

add : Int -> Int -> Int
add x y =
  x + y

type User = Anonymous | LoggedIn String

Notice that these declarations start with lower-case letters. Capitalization
makes a difference in Elm!
```

This should become available once Elm 0.19.1 is released. Thanks again for the report! The tweak in https://github.com/elm/compiler/commit/ef457e759bb50601fc680eb3f38946fa65dfe3a4 is based on your example!
